gcc/rust/parse/rust-parse-impl.h

```diff
@@ -2045,59 +2045,13 @@ Parser<ManagedTokenSource>::parse_macro_match ()
       case DOLLAR_SIGN: {
 	// have to do more lookahead to determine if fragment or repetition
 	const_TokenPtr t2 = lexer.peek_token (1);
+
+	if (token_id_is_keyword (t2->get_id ()) && t2->get_id () != CRATE)
+	  goto case_keyword;
+
 	switch (t2->get_id ())
 	  {
-	  case ABSTRACT:
-	  case AS:
-	  case ASYNC:
-	  case BECOME:
-	  case BOX:
-	  case BREAK:
-	  case CONST:
-	  case CONTINUE:
-	  case CRATE:
-	  case DO:
-	  case DYN:
-	  case ELSE:
-	  case ENUM_TOK:
-	  case EXTERN_TOK:
-	  case FALSE_LITERAL:
-	  case FINAL_TOK:
-	  case FN_TOK:
-	  case FOR:
-	  case IF:
-	  case IMPL:
-	  case IN:
-	  case LET:
-	  case LOOP:
-	  case MACRO:
-	  case MATCH_TOK:
-	  case MOD:
-	  case MOVE:
-	  case MUT:
-	  case OVERRIDE_TOK:
-	  case PRIV:
-	  case PUB:
-	  case REF:
-	  case RETURN_TOK:
-	  case SELF_ALIAS:
-	  case SELF:
-	  case STATIC_TOK:
-	  case STRUCT_TOK:
-	  case SUPER:
-	  case AUTO:
-	  case TRAIT:
-	  case TRUE_LITERAL:
-	  case TRY:
-	  case TYPE:
-	  case TYPEOF:
-	  case UNSAFE:
-	  case UNSIZED:
-	  case USE:
-	  case VIRTUAL:
-	  case WHERE:
-	  case WHILE:
-	  case YIELD:
+	  case_keyword:
```

Comment:
Jumping from outside the switch to inside the switch is not great IMO. I'm not in favor of this change. Do you have a testcase for the crate issue you mention? What about removing it from the list of cases in the switch statement?

Reply:
Since this pull request was closed and hasn't been reopened, the ```Files changed``` menu hasn't updated.

https://github.com/Rust-GCC/gccrs/compare/master...powerboat9:gccrs:fix-key

Reply:
Ah, my bad - can you reopen a pull-request? This will be easier. The new diff looks good to me, I would just like to see testcases

Reply:
I can't, I don't have the right permissions

Reply:
oh, that's weird. It tells me I can't either because the branch was force-pushed or recreated. Can you open another one or not at all?

Reply:
I could, but does it work now? I force pushed the original commit

Reply:
the button just unlocked for me lol. I reopened this one. I assume it's not in a valid state then?